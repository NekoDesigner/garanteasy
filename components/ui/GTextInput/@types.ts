/**
 * IGTextInputProps - Generated by Garanteasier CLI
 */

import { StyleProp, TextInputProps, TextStyle, ViewStyle } from "react-native";

export interface IGTextInputProps extends TextInputProps {
  containerStyle?: StyleProp<ViewStyle>;
  labelStyle?: StyleProp<TextStyle>;
  label: string;
  testID?: string;

}