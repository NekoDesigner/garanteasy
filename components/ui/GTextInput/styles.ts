/**
 * GTextInputStyles - Generated by Garanteasier CLI
 */

import { StyleSheet } from 'react-native';
import { COLORS } from '../../../constants';

const styles = StyleSheet.create({
  container: {
    marginVertical: 5,
  },
  label: {
    marginBottom: 5,
    color: COLORS.placeholder,
    fontSize: 14,
  },
  textInput: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 4,
    padding: 10,
    fontSize: 16,
    color: '#333',
    backgroundColor: '#fff',
  }
});

export default styles;
