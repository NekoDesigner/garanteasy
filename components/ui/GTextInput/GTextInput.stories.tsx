/**
 * GTextInput - Generated by Garanteasier CLI
 */

import React from "react";
import { View } from "react-native";
import GTextInput from "./index";

export default {
  title: 'GTextInput',
  component: GTextInput
};

export const Default = () => (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', padding: 16, gap: 24 }}>
      <GTextInput />
    </View>
);