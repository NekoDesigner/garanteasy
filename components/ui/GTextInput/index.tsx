/**
 * GTextInput - Generated by Garanteasier CLI
 */

import React from "react";
import { View, Text, TextInput } from "react-native";
import { IGTextInputProps } from "./@types";
import GTextInputStyles from "./styles";

const GTextInput: React.FC<IGTextInputProps> = ({style, containerStyle, labelStyle, testID = 'gtextinput', label, ...props}) => {
  return (
    <View style={[GTextInputStyles.container, containerStyle]} testID={testID}>
      <Text style={[GTextInputStyles.label, labelStyle]}>{label}</Text>
      <TextInput
        {...props}
        style={[GTextInputStyles.textInput, style]}
      />
    </View>
  );
};

export default GTextInput;
