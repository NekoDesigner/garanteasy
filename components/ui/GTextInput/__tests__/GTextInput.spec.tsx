/**
 * GTextInput - Generated by Garanteasier CLI
 */

import { render } from "@testing-library/react-native";
import React from "react";
import GTextInput from "../index";

describe("GTextInput Components", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

    test('renders correctly', () => {
      const tree = render(<GTextInput label="Test" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});