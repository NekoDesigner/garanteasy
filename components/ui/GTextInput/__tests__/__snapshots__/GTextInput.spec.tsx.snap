// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GTextInput Components renders correctly 1`] = `
<View
  style={
    [
      {
        "marginVertical": 5,
      },
      undefined,
    ]
  }
  testID="gtextinput"
>
  <Text
    style={
      [
        {
          "color": "#454545",
          "fontFamily": "Ubuntu",
          "fontSize": 14,
          "marginBottom": 5,
        },
        undefined,
      ]
    }
  >
    Test
  </Text>
  <TextInput
    style={
      [
        {
          "backgroundColor": "#fff",
          "borderColor": "#ccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "color": "#333",
          "fontSize": 16,
          "padding": 10,
        },
        undefined,
      ]
    }
  />
</View>
`;
